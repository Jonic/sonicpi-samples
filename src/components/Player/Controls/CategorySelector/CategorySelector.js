import './CategorySelector.scss'

import PropTypes from 'prop-types'
import React from 'react'
import Select from '../../../UI/Select/Select'

const CategorySelector = props => {
  const {
    categories,
    categorySelectorChangeHandler,
    currentCategory,
    isLoading,
  } = props

  return (
    <div className="c-category-selector">
      Category Selector:{' '}
      <Select
        className="c-category-selector__select"
        disabled={isLoading}
        onChange={categorySelectorChangeHandler}
        options={categories}
        value={currentCategory}
      />
    </div>
  )
}

CategorySelector.propTypes = {
  categories: PropTypes.array.isRequired,
  categorySelectorChangeHandler: PropTypes.func.isRequired,
  currentCategory: PropTypes.string.isRequired,
  isLoading: PropTypes.bool.isRequired,
}

export default CategorySelector
