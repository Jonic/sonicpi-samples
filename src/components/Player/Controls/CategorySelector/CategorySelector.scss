@import '../../../../styles/component-helper';

.c-CategorySelector {}
