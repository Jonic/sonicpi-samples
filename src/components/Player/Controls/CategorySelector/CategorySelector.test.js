import React from 'react'
import { shallow } from 'enzyme'
import CategorySelector from './CategorySelector'

const categories = []
const categorySelectorChangeHandler = jest.fn()
const currentCategory = ''
const isLoading = false

let wrapper

describe('<CategorySelector />', () => {
  beforeEach(() => {
    wrapper = shallow(
      <CategorySelector
        categories={categories}
        categorySelectorChangeHandler={categorySelectorChangeHandler}
        currentCategory={currentCategory}
        isLoading={isLoading}
      />
    )
  })

  describe('Renders...', () => {
    it('without crashing', () => {
      expect(wrapper.find('.c-CategorySelector')).toBePresent()
    })

    it('matches its snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
