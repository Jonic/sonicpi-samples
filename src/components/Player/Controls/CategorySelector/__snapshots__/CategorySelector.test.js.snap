// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategorySelector /> Renders... matches its snapshot 1`] = `
<div
  className="c-category-selector"
>
  Category Selector:
   
  <Select
    className="c-category-selector__select"
    disabled={false}
    onChange={[MockFunction]}
    options={Array []}
    value=""
  />
</div>
`;
